dotCMS/src/main/java/com/dotcms/security/secret/ServiceIntegrationAPI.java

```diff
@@ -0,0 +1,58 @@
+package com.dotcms.security.secret;
+
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.Config;
+import com.liferay.portal.model.User;
+import io.vavr.control.Try;
+import java.util.List;
+import java.util.Optional;
+
+public interface ServiceIntegrationAPI {
+
+    /**
+     * This returns json object read from the secret store that contains the service integration configuration and secret
+     * @param serviceKey
+     * @return
+     */
+     Optional<ServiceIntegrationBean> getIntegrationForService(String serviceKey, User user) throws DotDataException;
+
+    /**
+     *
+     * @param bean
+     */
+    void registerServiceIntegration(final ServiceIntegrationBean bean, User user)
+            throws DotDataException, DotSecurityException;
+
+    /**
+     *
+     * @param serviceKey
+     */
+    void deleteServiceIntegration(String serviceKey, User user)
+            throws DotDataException, DotSecurityException;
+
+    /**
+     * This method should read the yml file service definition
+     * @return
+     */
+    List<ServiceIntegrationBean> getAvailableServiceDescriptors(User user);
+
+
+    enum INSTANCE {
```

Comment:
it is a weird name

Reply:
yeah I agree

Reply:
I don't disagree, it just makes calling the singleton look ok, e.g:
`SecretsStore.INSTANCE.get()`

It could be anything though...

Reply:
that's true too.